job-init/.bashrc.github.inc.sh

```diff
@@ -0,0 +1,317 @@
+github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==
+EOF
+}
+
+hawk.setup.git() {
+  git config --global user.name "GitHub Actions"
+  git config --global user.email "github-actions@hawk.ai"
+}
+
+hawk.job-init() {
+  # Make sure locale is set to en_US.UTF-8
+  # https://hawkai.atlassian.net/browse/SRE-690
+  # https://hawkai.atlassian.net/browse/SRE-734
+  hawk.setup.locale
+
+  # Make sure yj is installed, which may be not the case
+  hawk.setup.yj
+
+  # Setup known hosts 
+  hawk.setup.ssh-known-hosts
+
+  # Prepar git for potential pushes
+  hawk.setup.git
+
+  BUILD_BRANCH=$(echo ${GITHUB_REF_NAME} | sed 's,\/,-,g; s,\#,,g')
+  GIT_SHA_SHORT=${GITHUB_SHA:0:7}
+  if [[ ${GITHUB_REF_TYPE} == "tag" ]]; then
+    IMAGE_TAG=${BUILD_BRANCH}
+  else
+    IMAGE_TAG=${BUILD_BRANCH}-${GIT_SHA_SHORT}
+  fi
+
+  cat << EOF | tee -a ${GITHUB_ENV}
+HAWK_METADATA_REPO=hawk-ai-aml/github-actions
+HAWK_METADATA_SUBPATH=workflow-init/profile
+HAWK_METADATA_DEFAULT_REF=master
+HAWK_METADATA_DEFAULT_PROFILE=hawk
+
+HAWK_BUILD_BRANCH=${BUILD_BRANCH}
+HAWK_IMAGE_TAG=${IMAGE_TAG}
+HAWK_GIT_SHA_SHORT=${GIT_SHA_SHORT}
+EOF
+}
+
+################################################################################
+# Working with metadata and profiles
+################################################################################
+
+hawk.get-metadata-json() {
+  local repo=${HAWK_METADATA_REPO}
+  local subpath=${HAWK_METADATA_SUBPATH}
+  local ref=${HAWK_METADATA_DEFAULT_REF}
+
+  local profile=${1:-${HAWK_METADATA_DEFAULT_PROFILE}}
+
+  [[ ! -z "${profile}" ]] || hawk.die "metadata profile should not be empty"
+  [[ ! -z "${ref}" ]] || hawk.die "metadata ref should be not empty"
+  [[ ! -z "${repo}" ]] || hawk.die "metadata repo should be not empty"
+  [[ ! -z "${subpath}" ]] || hawk.die "metadata subpath should be not empty"
+
+  local metadata_url="https://raw.githubusercontent.com/${repo}/${ref}/${subpath}/${profile}.yml"
+  local metadata_workspace_path="${GITHUB_WORKSPACE}/.hawk/profile/${profile}.yml"
+  
+  # if there is profile folder in the repo we get profiles from there, otherwise we fetch it from github-actions
+  # We pass it through both yj and jq to make sure there are no unexpected basic parsing errors
+  if [[ -f ${metadata_workspace_path} ]]; then
+    cat ${metadata_workspace_path} | yj -y | jq -Mcr .
+  else
+    curl --silent --fail --location --show-error ${metadata_url} | yj -y | jq -Mcr .
+  fi
+}
```

Comment:
hiding structure is possible to do here, as discussed yesterday via 2 possiblities
1. easier - to fetch it from protected place
2. better for maintenance - to spread big metadata yaml as a yaml pieces inside actual repositories, so developers can update both gradle and these yaml fields in one go